src/main/kotlin/org/scoverage/ScoverageExtension.kt

```diff
@@ -0,0 +1,86 @@
+package org.scoverage
+
+import groovy.lang.Closure
+import org.gradle.api.Project
+import org.gradle.api.plugins.JavaPlugin
+import org.gradle.api.plugins.scala.ScalaPlugin
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.provider.Property
+import java.io.File
+import java.math.BigDecimal
+
+/**
+ * Defines a new SourceSet for the code to be instrumented.
+ * Defines a new Test Task which executes normal tests with the instrumented classes.
+ * Defines a new Check Task which enforces an overall line coverage requirement.
+ */
+abstract class ScoverageExtension(val project: Project) {
```

Comment:
Similarly to tasks, you can define all properties here as `abstract`, removing the need to invoke `project` directly. You can even make this class into an interface, and configure it in the plugin logic.